### PR TITLE
ADEN-3902 Enable Ooyala ads on mobile

### PR DESCRIPTION
**Pull request by fraszczakszymon**
Enable ads in Ooyala player (using new VastBuilder)

Wait for https://github.com/Wikia/app/pull/11315 before releasing 

## Links

* http://wikia-inc.atlassian.net/browse/ADEN-3902

### CODE REVIEW

File: front/main/app/services/ads-high-impact.js [hunk 1]
```diff
@@ -34,6 +35,6 @@ export default Ember.Service.extend({
 	},
 
 	loadFloorAdhesionWhenPossible() {
-		this.get('adsState').runWhenScriptLoaded(this.loadFloorAdhesion, this);
+		Ads.getInstance().onReady(this.loadFloorAdhesion, this);
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: how about using `adsState.get('module')`? maybe it's better to access Ads module only from one place in ember - so there is only one service truly dependent on it.

File: front/main/app/services/ads-high-impact.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import Ember from 'ember';
+import Ads from 'common/modules/ads';
```
Comment (bognix) [inline front/main/app/services/ads-high-impact.js]: remove not used import

File: front/main/app/services/ads-state.js [hunk 3]
```diff
@@ -7,17 +7,9 @@ export default Ember.Service.extend({
 	siteHeadOffset: 0,
```
Comment (bognix) [inline front/main/app/services/ads-state.js]: I'm quite sure this service (this property) has unit tests. Could you add a test case to check what happens when userId is returned?

File: front/common/modules/video-players/ooyala.js [hunk 4]
```diff
@@ -43,7 +38,26 @@ export default class OoyalaPlayer extends BasePlayer {
 	 * @returns {void}
 	 */
 	createPlayer() {
-		window.OO.Player.create(this.containerId, this.params.videoId, this.params);
+		Ads.getInstance().onReady(function () {
+			const vastUrl = Ads.getInstance().buildVastUrl();
+
+			this.params = $.extend(this.params, {
+				onCreate: (...args) => {
+					return this.onCreate.apply(this, args);
```
Comment (fraszczakszymon) [inline front/common/modules/video-players/ooyala.js]: We're extending `this.params` object with `onCreate` method

Comment (nandy-andy) [inline front/common/modules/video-players/ooyala.js]: Yep, and I'm guessing it's a kind of an event handler used by the player.

File: front/common/modules/ads.js [hunk 5]
```diff
@@ -129,6 +138,19 @@ class Ads {
...
 	}
 
+	/**
+	 * Build VAST url for video players
+	 *
+	 * @returns {string}
+	 */
+	buildVastUrl() {
+		if (!this.vastBuilder) {
+			return '';
```
Comment (fraszczakszymon) [inline front/common/modules/ads.js]: Addressed

File: front/common/modules/video-players/ooyala.js (hunk 4 above)
Comment (fraszczakszymon) [inline front/common/modules/video-players/ooyala.js]: Addressed

File: front/common/modules/video-players/ooyala.js [hunk 6]
```diff
@@ -43,7 +38,26 @@ export default class OoyalaPlayer extends BasePlayer {
...
+				}
+			});
+
+			if (!this.params.noAds) {
+				this.params = $.extend(this.params, {
+					'google-ima-ads-manager': {
+						adTagUrl: vastUrl,
+						showInAdControlBar: true
```
Comment (fraszczakszymon) [inline front/common/modules/video-players/ooyala.js]: Addressed

File: front/common/tests/unit/modules/video-players/ooyala.js [hunk 7]
```diff
@@ -18,11 +18,15 @@ QUnit.module('mercury/modules/video-players/ooyala', function (hooks) {
 
 	hooks.beforeEach(function () {
 		var loadStub = sinon.stub().callsArg(1),
+			adsExports = {},
 			baseExports = {},
 			exports = {};
 
+		require.entries['common/modules/ads'].callback(adsExports, loadStub);
```
Comment (fraszczakszymon) [inline front/common/tests/unit/modules/video-players/ooyala.js]: Addressed

File: front/main/app/services/ads-state.js (hunk 3 above)
Comment (fraszczakszymon) [inline front/main/app/services/ads-state.js]: Addressed

**Pull request merged**